### DISCUSSION TITLE
How to set access to settings page ?

### DISCUSSION THREAD

**Discussion by sh4msi** [filamentphp/filament#5231] [Help] [answered]
How can I control the access to settings (filament/spatie-laravel-settings-plugin) like resources?

```
    public static function can(): bool
    {
        return auth()->user()->isAdmin();
    }
```

Of course, I know that I can use `mount()` to apply this limitation
But I don't know how to control `$shouldRegisterNavigation `

**Comment by ManojKiranA** [comment, marked as answer]
You can use `shouldRegisterNavigation` and `mount` together like this

```
protected static function shouldRegisterNavigation(): bool
{
    return auth()->user()->isAdmin();
}

public function mount()
{
    abort_unless(auth()->user()->isAdmin());
}
```
